library/core/src/option.rs

```diff
@@ -938,6 +940,38 @@ impl<T> Option<T> {
         self
     }
 
+    /// Calls the provided closure if [`None`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(result_option_inspect)]
+    ///
+    /// let v = vec![1, 2, 3, 4, 5];
+    ///
+    /// // prints nothing
+    /// let a: Option<&usize> = v.get(3).inspect_none(|| println!("index out of bound!"));
+    /// assert_eq!(a, Some(&4));
+    ///
+    /// // prints "index out of bound!"
+    /// let b: Option<&usize> = v.get(5).inspect_none(|| println!("index out of bound!"));
+    /// assert_eq!(b, None);
+    /// ```
+    #[inline]
+    #[unstable(feature = "result_option_inspect", issue = "91345")]
+    #[rustc_const_unstable(feature = "const_option_ext", issue = "91930")]
+    pub const fn inspect_none<F>(self, f: F) -> Self
+    where
+        F: ~const FnOnce(),
+        F: ~const Drop,
```

Comment:
I don't really understand the details of `const_option_ext`. Please verify that this is correct.

Reply:
This is correct. I think you are referring to `const_trait_impl`? Because `const_option_ext` is just a library tracking issue.

It is correct to bound `~const Drop` on generic types that *could* get dropped, in this case it will be dropped when the option is `Some`.

Reply:
Okay good to know. Thank you.